File: src/main/java/net/flyingfishflash/loremlist/domain/lrmitem/LrmItemService.java
package net.flyingfishflash.loremlist.domain.lrmitem;

import java.util.List;
import java.util.UUID;
import net.flyingfishflash.loremlist.domain.ServiceResponse;
import net.flyingfishflash.loremlist.domain.lrmitem.data.LrmItemDeleted;

public interface LrmItemService {
  ServiceResponse<Long> countByOwner(String owner);

  ServiceResponse<LrmItemDeleted> deleteByOwner(String owner);

  ServiceResponse<LrmItemDeleted> deleteByOwnerAndId(
      UUID id, String owner, boolean removeListAssociations);

  ServiceResponse<List<LrmItem>> findByOwner(String owner);

  ServiceResponse<LrmItem> findByOwnerAndId(UUID id, String owner);

  ServiceResponse<List<LrmItem>> findByOwnerAndHavingNoListAssociations(String owner);

  ServiceResponse<List<LrmItem>> findByOwnerAndHavingNoListAssociations(String owner, UUID listId);

  void patchName(LrmItem patchedLrmItem);

  void patchDescription(LrmItem patchedLrmItem);
}

File: src/main/kotlin/net/flyingfishflash/loremlist/domain/lrmitem/LrmItemService.kt
package net.flyingfishflash.loremlist.domain.lrmitem

import net.flyingfishflash.loremlist.domain.ServiceResponse
import net.flyingfishflash.loremlist.domain.lrmitem.data.LrmItemDeleted
import java.util.UUID

interface LrmItemService {
  fun countByOwner(owner: String): ServiceResponse<Long>
  fun deleteByOwner(owner: String): ServiceResponse<LrmItemDeleted>
  fun deleteByOwnerAndId(id: UUID, owner: String, removeListAssociations: Boolean): ServiceResponse<LrmItemDeleted>
  fun findByOwner(owner: String): ServiceResponse<List<LrmItem>>
  fun findByOwnerAndId(id: UUID, owner: String): ServiceResponse<LrmItem>
  fun findByOwnerAndHavingNoListAssociations(owner: String): ServiceResponse<List<LrmItem>>
  fun findByOwnerAndHavingNoListAssociations(owner: String, listId: UUID): ServiceResponse<List<LrmItem>>
  fun patchName(patchedLrmItem: LrmItem)
  fun patchDescription(patchedLrmItem: LrmItem)
}
